Выполнено Задание1 и ответы на первые 3 вопроса

DmtrPn opened the pull request.

DmtrPn committed:
Выполнено Задание1 и ответы на первые 3 вопроса

DmtrPn committed:
Задание полностью выполнено

ivan-kleshnin commented on common/types/PerfomanceComment.js:
```diff
@@ -0,0 +1,37 @@
+
+let PerfomanceComment = T.struct({
+  id: Id,
+  perfimanceId: Id,
```
Опечатка

ivan-kleshnin commented on common/types/PerfomanceComment.js:
```diff
@@ -0,0 +1,37 @@
+let PerfomanceComment = T.struct({
+  id: Id,
+  perfimanceId: Id,
+  formUserId: Id,
```
Это внешний ключ чего? Формы? Внешние ключи всегда относятся к таблицам, а формы не соотв. таблицам. Или я не понял идею или тут ошибка.

DmtrPn replied:
Посмотрев видео, решил также добавить в комментариях возможность пользователям с ролями `actor` и `director` отвечать на комментарии других пользователей. 
Сейчас еще и опечатку заметил, должно быть:
`fromUserId ` - ключ пользователя, написавшего комментарий
`toUserId` - ключ пользователя на чей комментарий был получен ответ

ivan-kleshnin commented on common/types/User.js:
```diff
@@ -0,0 +1,42 @@
+  countOfPerfomance: T.Number
+}, "UserForm")
+
+let makeUser = (data) => {
```
Круто, что ты сделал генерацию. По моей идее это требовалось позже, но в видео показал почему-то :|

ivan-kleshnin commented on common/types/User.js:
```diff
@@ -0,0 +1,42 @@
+let T = require("tcomb")
+let {Id, makeId} = require("./Id")
+let Faker = require("faker")
+let Slug = require("speakingurl")
```
Если это просто функция, то их обычно импортируют в нижнем регистре.
`let slug = ...`
Я импортирую с капитализацией либо классы, либо модули-объекты, т.е. содержащие N функций.

ivan-kleshnin commented on common/types/User.js:
```diff
@@ -0,0 +1,42 @@
+  userName: T.String,
+  inclusionDate: T.Date,
+  exclusionDate: T.Date,
+  countOfPerfomance: T.Number
```
Эта структура соответствует Форме (т.е. веб-форме, кот. заполняет пользователь). Тут не может быть статистических и мета-данных типа дат, кол-ва комментариев и т.п. 

Тут, видимо, должно быть только `userName`.

ivan-kleshnin commented on common/types/User.js:
```diff
@@ -0,0 +1,42 @@
+  countOfPerfomance: T.Number
+}, "User")
+
+let UserForm = T.struct({
```
Эта структура соответствует Форме (т.е. веб-форме, кот. заполняет пользователь). Тут не может быть статистических и мета-данных типа дат, кол-ва комментариев и т.п. 

Тут, видимо, должно быть только `userName`.

DmtrPn committed:
Исправлены User.js (в UserForm оставлен только userName и slug без
капитализации)
PerfomanceComment.js (исправлены опечатки и добавлен комментарий к
fromUserId и toUserId)

ivan-kleshnin commented on common/types/PerfomanceComment.js:
```diff
@@ -5,9 +5,10 @@ let Chance = require("chance")
-  formUserId: Id,
-  toUserId: Id,
+  perfomanceId: Id,
+  fromUserId: Id, // кто написал комментарий
```
So it was a typo: `form` -> `from`. I was confused by it.
Russian vs Eng mix again. Please replace with eng.
And let's communicate here in English!
Agree?

DmtrPn replied:
Agree

DmtrPn committed:
comments translated to english
